composition/src/normalization/normalization-factory.ts

```diff
@@ -530,13 +520,15 @@ export class NormalizationFactory {
       return;
     }
     const isEntity = isObjectLikeNodeEntity(node);
-    upsertExtensionWithFieldsDataByNode(
+    addExtensionWithFieldsDataByNode(
       this.parentExtensionDataByTypeName,
       node,
       this.errors,
       this.directiveDefinitionByDirectiveName,
       this.handledRepeatedDirectivesByHostPath,
       isEntity,
+      isRootType,
+      this.subgraphName,
     );
     // TODO re-assess this line
```

Comment:
Still a todo?

Reply:
Yes. for now. It's to remind me to check the logic here shouldn't change after implementing entity interfaces.